--- mod_auth_radius-2.0.c	2012-03-06 10:16:22.233506682 -0500
+++ mod_auth_radius-2.0.c.oden	2012-03-06 10:16:52.355506813 -0500
@@ -680,7 +680,7 @@ make_cookie(request_rec *r, time_t expir
    * benefit here.
    */  
   apr_snprintf(one, COOKIE_SIZE, "%s%s%s%s%s%08x", scr->secret,
-	      r->user, passwd, c->remote_ip, hostname, expires);
+	      r->user, passwd, c->client_ip, hostname, expires);
 
   /* if you're REALLY worried about what's going on */
 
@@ -688,7 +688,7 @@ make_cookie(request_rec *r, time_t expir
   ap_log_error(APLOG_MARK, APLOG_NOERRNO | APLOG_DEBUG, 0, r->server," secret     = %s\n", scr->secret);
   ap_log_error(APLOG_MARK, APLOG_NOERRNO | APLOG_DEBUG, 0, r->server," user       = %s\n", r->user);
   ap_log_error(APLOG_MARK, APLOG_NOERRNO | APLOG_DEBUG, 0, r->server," passwd     = %s\n", passwd);
-  ap_log_error(APLOG_MARK, APLOG_NOERRNO | APLOG_DEBUG, 0, r->server," remote ip  = %s\n", c->remote_ip);
+  ap_log_error(APLOG_MARK, APLOG_NOERRNO | APLOG_DEBUG, 0, r->server," remote ip  = %s\n", c->client_ip);
   ap_log_error(APLOG_MARK, APLOG_NOERRNO | APLOG_DEBUG, 0, r->server," hostname   = %s\n", hostname);
   ap_log_error(APLOG_MARK, APLOG_NOERRNO | APLOG_DEBUG, 0, r->server," expiry     = %08x\n", expires);
 #endif
